Assets/Plugins/IOS/RadarUnityBridge.m

```
@@ -18,12 +18,13 @@
 
     - (void)didUpdateToken:(RadarVerifiedLocationToken *)token {
         if (_tokenUpdatedCallback && token) {
-            const char *tokenString = [token.token UTF8String];
-            BOOL passed = token.passed;
-            long expiresAt = [token.expiresAt timeIntervalSince1970] * 1000; // Convert seconds to milliseconds
-            int expiresIn = token.expiresIn;
-            
-            _tokenUpdatedCallback(tokenString, passed, expiresAt, expiresIn);
+            NSDictionary *dict = [token dictionaryValue];
+            NSError *error;
+            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:dict options:0 error:&error];
+            if (!error) {
+                NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+                _tokenUpdatedCallback([jsonString UTF8String]);
+            }
```

is there a mechanism to surface the error? It feels wrong to just discard it and fail silently.

---

good catch. let's save it for the refactor